### PR TITLE
Tickets/pipe2 d 147

### PR DESCRIPTION
**Pull request by ziggyman**

### CODE REVIEW

File: tests/Mapper.py [hunk 1]
```diff
@@ -45,34 +55,63 @@ def tearDown(self):
...
                 fileNameFormat = 'PF%1s%1s%06d%1d%1d.fits'
-                fileName = fileNameFormat % (site, category, visit, spectrograph, armNum)
+                fileName = fileNameFormat % (self.site, self.category, self.visit, spectrograph, armNum)
                 dirName = os.path.dirname( self.rawFiles )
                 fileNameWithDir = os.path.join( dirName, fileName )
                 info = pfsParseTask.getInfo( fileNameWithDir )
-                self.assertTrue(info[0]['site'] == site)
-                self.assertTrue(info[0]['category'] == category)
-                self.assertTrue(info[0]['visit'] == visit)
+                self.assertTrue(info[0]['site'] == self.site)
```
Comment (RobertLuptonTheGood) [inline tests/Mapper.py]: Use `assertEqual(a, b)`, not `assertTrue(a == b)` wherever possible (better error messages)

File: tests/Mapper.py [hunk 2]
```diff
@@ -27,16 +29,24 @@ def setUp(self):
...
-        
+
         self.rawFiles = os.path.join(drpStellaDataDir,"tests/data/raw/*.fits")
         self.regFile = os.path.join(self.butlerDir,'registry.sqlite3')
-        self.dataId = {'visit': '4', 'ccd': 1}
+
+        self.site = 'S'
+        self.category = 'A'
+        self.visit = 4
+        self.arms = ['b', 'r', 'n', 'm']
```
Comment (RobertLuptonTheGood) [inline tests/Mapper.py]: If you set `self.arms` to `[None, 'b', ...]` you won't need all the `armNum-1` expressions elsewhere.

File: tests/Mapper.py [hunk 3]
```diff
@@ -45,34 +55,63 @@ def tearDown(self):
...
-        for armNum in np.arange(minArmNum, maxArmNum + 1, 1):
-            for spectrograph in np.arange(minSpectrographNum, maxSpectrographNum + 1, 1):
-                ccd = spectrograph - 1
-                if int(armNum) == minArmNum + 1 or int(armNum) == minArmNum + 3:
-                    ccd += 4
-                elif int(armNum) == minArmNum + 2:
-                    ccd += 8
+        mapper = PfsMapper(
+            root=os.path.join(lsst.utils.getPackageDir('drp_stella_data'),'tests/data/PFS')
+        )
```
Comment (RobertLuptonTheGood) [inline tests/Mapper.py]: What happens if `drp_stella_data` isn't set up?

File: tests/Mapper.py [hunk 4]
```diff
@@ -45,34 +55,63 @@ def tearDown(self):
...
-                self.assertTrue(info[0]['filter'] == arms[armNum-1])
+                self.assertTrue(info[0]['arm'] == arm)
+                self.assertTrue(info[0]['filter'] == arm)
                 self.assertTrue(info[0]['ccd'] == ccd)
 
+    def testDetectorId(self):
+        butler = dafPersist.Butler(self.butlerDir)
+        mapper = butler.getMapperClass(self.butlerDir)(root=self.butlerDir)
+        camera = mapper.camera
+        blues=[0,3,6,9]
```
Comment (RobertLuptonTheGood) [inline tests/Mapper.py]: a comment about what these arrays mean would be appreciated

File: tests/Mapper.py (hunk 2 above)
Comment (ziggyman) [inline tests/Mapper.py]: Done

File: tests/Mapper.py (hunk 3 above)
Comment (ziggyman) [inline tests/Mapper.py]: fixed to run tests only if drp_stella_data is setup, show warning message on screen if not setup

File: tests/Mapper.py (hunk 1 above)
Comment (ziggyman) [inline tests/Mapper.py]: Done

File: tests/Mapper.py (hunk 4 above)
Comment (ziggyman) [inline tests/Mapper.py]: Added comment

**Comment by ziggyman** [comment]
Merged into master

**Pull request merged**